pkg/healthcheck/healthcheck_output.go

```diff
@@ -48,20 +52,200 @@ func (cr CheckResults) RunChecks(observer CheckObserver) bool {
+		}
+
+		if err == nil {
+			if isatty.IsTerminal(os.Stdout.Fd()) {
+				spin.Suffix = fmt.Sprintf(" Running %s extension check", extension)
+				spin.Color("bold") // this calls spin.Restart()
+			}
+			plugin := exec.Command(path, args...)
+			var stdout, stderr bytes.Buffer
+			plugin.Stdout = &stdout
+			plugin.Stderr = &stderr
+			plugin.Run()
+			extensionResults, err := parseJSONCheckOutput(stdout.Bytes())
+			spin.Stop()
+			if err != nil {
+				command := fmt.Sprintf("%s %s", path, strings.Join(args, " "))
+				if len(stderr.String()) > 0 {
+					err = errors.New(stderr.String())
+				} else {
+					err = fmt.Errorf("invalid extension check output from \"%s\" (JSON object expected):\n%s\n[%s]", command, stdout.String(), err)
+				}
+				results.Results = append(results.Results, CheckResult{
+					Category:    CategoryID(extensionCmd),
+					Description: fmt.Sprintf("Running: %s", command),
+					Err:         err,
+					HintURL:     DefaultHintBaseURL + "extensions",
+				})
+				success = false
+			} else {
+				results.Results = append(results.Results, extensionResults.Results...)
+			}
+		}
+
+		var containsErr bool
+		for _, res := range results.Results {
+			if res.Err != nil {
+				containsErr = true
+				break
+			}
+		}
+
+		if output != JSONOutput && i < len(extensions) {
+			// add a new line to space out each check output
+			fmt.Fprintln(wout)
+			if output == ShortOutput && !containsErr {
+				fmt.Fprintln(wout, string(results.Results[0].Category))
+				fmt.Fprintln(wout, strings.Repeat("-", len(string(results.Results[0].Category))))
+			}
+		}
+
+		extensionSuccess := RunChecks(wout, werr, results, output)
+		if !extensionSuccess {
+			success = false
+		}
+	}
+
+	return success
+}
+
 // RunChecks runs the checks that are part of hc
 func RunChecks(wout io.Writer, werr io.Writer, hc Runner, output string) bool {
 	if output == JSONOutput {
 		return runChecksJSON(wout, werr, hc)
 	}
+
+	if output == ShortOutput {
+		return runChecksShort(wout, hc)
+	}
+
 	return runChecksTable(wout, hc)
 }
 
-func runChecksTable(wout io.Writer, hc Runner) bool {
+func runChecksShort(wout io.Writer, hc Runner) bool {
 	var lastCategory CategoryID
+
 	spin := spinner.New(spinner.CharSets[9], 100*time.Millisecond)
 	spin.Writer = wout
 
 	prettyPrintResults := func(result *CheckResult) {
+		spin.Stop()
+		if result.Retry {
+			if isatty.IsTerminal(os.Stdout.Fd()) {
+				spin.Suffix = fmt.Sprintf(" %s", result.Err)
+				spin.Color("bold") // this calls spin.Restart()
+			}
+			return
+		}
+
+		if result.Err == nil {
+			return
+		}
+
+		status := failStatus
+		if result.Warning {
+			status = warnStatus
+		}
+
+		if lastCategory != result.Category {
+			if lastCategory != "" {
+				fmt.Fprintln(wout)
+			}
+
+			fmt.Fprintln(wout, result.Category)
+			fmt.Fprintln(wout, strings.Repeat("-", len(result.Category)))
+
+			lastCategory = result.Category
+		}
+
+		fmt.Fprintf(wout, "%s %s\n", status, result.Description)
+		if result.Err != nil {
```

Comment:
I'm a bit confused about this but thought I might ask anyway.

Since on line 199 we are return if the contained `Err` is nil, is it safe to assume that we _will_ have an error here so we can skip the If conditional? Likewise, I see that on line 203 we set the status to a fail status (preemptively) which makes me think this can either be a warning or a failure? 

In case we can safely assume the error will not be nil we might be able to reduce the nesting by getting rid of the outermost if branch?
```suggestion
	
```

Let me know if I'm wrong here, not very familiar with the healthchecks

Reply:
Good catch! Line 220 needs to assume that `result.Err` is not nil since we bail out early on 199. This was an oversight on my part.